### PR TITLE
Add documentation for what functions invoked in model.train as in model.predict

### PR DESCRIPTION
**Pull request by bilal-aamer**
#148 
Updated the docs of the `unionml.model.Model.train` method to show what functions are invoked as done in the `model.predict` method.

### CODE REVIEW

File: unionml/model.py [hunk 1]
```diff
@@ -514,15 +514,19 @@ def train(
 
         :param hyperparameters: a dictionary mapping hyperparameter names to values. This is passed into the
             ``init`` callable to initialize a model object.
-        :param loader_kwargs: key-word arguments to pass to the registered :meth:`unionml.Dataset.loader` function.
+        :param loader_kwargs: key-word arguments to pass to the registered
```
Comment (cosmicBboy) [inline unionml/model.py:517]: hi @bilal-aamer wanted to clarify the updates needed for https://github.com/unionai-oss/unionml/issues/148.

It needs to list the tasks that are invoked in the training workflow.

Can you revert the changes that you made below and add a paragraph below the argument docstrings, something like:

```
The train method invokes an execution graph that composes together the following functions to train and evaluate a model:

- :meth:`unionml.Dataset.reader`
- :meth:`unionml.Dataset.loader`
- :meth:`unionml.Dataset.splitter`
- :meth:`unionml.Dataset.parser`
- :meth:`unionml.Model.trainer`
- :meth:`unionml.Model.predictor`
- :meth:`unionml.Model.evaluator`
```

Comment (bilal-aamer) [reply unionml/model.py:517]: Sure will do

Comment (bilal-aamer) [reply unionml/model.py]: Done! @cosmicBboy ! Please Review!

**Review by cosmicBboy — APPROVED** [review]
thanks @bilal-aamer !

**Comment by bilal-aamer** [comment]
No, thank you @cosmicBboy 👍🏼
